te3/tutorials/data-security/data-security-testing.md

```diff
@@ -0,0 +1,81 @@
+---
+# Testing Data Security with Impersonation
+
+
+![Data Security Visual Abstract](~/images/data-security/data-security-testing-visual-abstract.png)
+
+
+---
+
+__DAX Queries__, __Pivot Grid__, or __Preview Data__ enable testing of Data Security in Tabular Editor. It is recommended to *always* test Data Security with any changes to configuration, to mitigate risk of incorrect RLS/OLS implementation and the consequences thereof.
+
+> [!IMPORTANT]
+> Testing Data Security with Impersonation using Tabular Editor 3 is limited to dataset hosted in the Power BI Datasets service. TE3 Desktop Licenses cannot benefit from this feature. This is because roles are assigned in the Power BI Service.
+
+---
+
+
+- [__About Data Security and RLS/OLS:__](data-security-about.md) A functional overview of <span style="color:#01a99d">RLS</span> & <span style="color:#8d7bae">OLS</span>.
+- [__Modify/Setup an RLS Configuration:__](data-security-setup-rls.md) How to configure <span style="color:#01a99d">RLS</span> in a dataset.
+- [__Modify/Setup an OLS Configuration:__](data-security-setup-ols.md) How to configure <span style="color:#8d7bae">OLS</span> in a dataset.
+- __Testing RLS/OLS with Impersonation (This Article):__ How to easily validate Data Security with Tabular Editor.
+
+---
+
+## Testing with Impersonation
+
+__Data Security can be easily tested using _Impersonation_ in Tabular Editor 3.__ Impersonation is a feature that lets you view a query result as a model Role or User. It is similar to the _'View As Role...'_ feature in the Power BI service, with two key differences:
+
+1. The User requires __dataset Build permissions__ in addition to Role assignment & Dataset Read access. 
+2. Any query can be executed; it is not limited to available reports.
+
+ This is valuable, as it lets the developer run defined tests to see how the result would be viewed by any user with Build permissions. This helps ensure that even for complex queries and DAX expressions, the Data Security works as expected, and users only see what they should see.
+
+
+> [!IMPORTANT]
+> Ensure that Build permissions are not provisioned by providing users Workspace Roles (Contributor, Member, Admin), as these roles have __Write__ permissions to the dataset and thus bypass Data Security; the testing will appear to not work, even if it's configured correctly. 
+
+
+<figure style="padding-top: 15px;">
+  <img class="noscale" src="~/images/data-security/data-security-impersonation-demo.gif" alt="Data Security Create Role" style="width: 550px;"/>
+  <figcaption style="font-size: 12px; padding-top: 10px; padding-bottom: 15px; padding-left: 75px; padding-right: 75px; color:#00766e"><strong>Figure 1:</strong> A demonstration of RLS testing in Tabular Editor using impersonation. Shown is testing with (A) Data Preview, (B) DAX Queries and (C) Pivot Grid.</figcaption>
+</figure>
+
+---
+
+## How to Test with Impersonation
+To test with impersonation, follow the below steps:
```

Comment:
For AS I honestly can't say since I don't use it. The only difference though I believe is that you assign roles within the metadata. Role level access gives dataset read permissions, and the permissions within the role for objects or tables define the RLS/OLS. I don't know if there's a separate thing in AS for provisioning artifact access though.

Reply:
I think we either need to specify that the bellow is only for Power BI or add a section on Analysis Services since we cater to both. Thoughts @otykier

Reply:
Yeah, we should make it clear that the configuration/access outlines below is specifically for Power BI dataset. When testing RLS/OLS on an Analysis Services model, all that is required, is that the developer has server admin permission on the AS instance.

Reply:
I propose that we just mention line 65-66 are PBI specific; I think the rest of this section is generic enough to apply to both PBI and AAS; the Workspace Role and Buildl Permission points are just specific notes for Power BI.

Reply:
Agreed